block/local/adapter_test.go

```diff
@@ -141,3 +142,56 @@ func TestLocalCopy(t *testing.T) {
 		t.Errorf("expected to read \"%s\" as written, got \"%s\"", contents, string(got))
 	}
 }
+
+func TestRemove(t *testing.T) {
+	a := makeAdapter(t)
+	const contents = "foo bar baz quux"
+
+	obj1 := makePointer("README")
+	err := a.Put(obj1, 0, strings.NewReader(contents), block.PutOpts{})
+	testutil.MustDo(t, "Put", err)
+	err = a.Remove(obj1)
+	testutil.MustDo(t, "Remove", err)
+
+	root := a.Path()
+	tree1 := dumpPathTree(t, root)
+	if diff := deep.Equal(tree1, []string{""}); diff != nil {
+		t.Fatalf("Found diff after remove: %s", diff)
+	}
+
+	obj2 := makePointer("src/tools.go")
+	err = a.Put(obj2, 0, strings.NewReader(contents), block.PutOpts{})
+	testutil.MustDo(t, "Put", err)
+	err = a.Remove(obj2)
+	testutil.MustDo(t, "Remove", err)
+	tree2 := dumpPathTree(t, root)
+	if diff := deep.Equal(tree2, []string{""}); diff != nil {
+		t.Fatalf("Found diff after remove: %s", diff)
+	}
+
+	obj3 := makePointer("a/b/c/d.txt")
+	err = a.Put(obj3, 0, strings.NewReader(contents), block.PutOpts{})
+	testutil.MustDo(t, "Put", err)
+	err = a.Remove(obj3)
+	testutil.MustDo(t, "Remove", err)
+	tree3 := dumpPathTree(t, root)
+	if diff := deep.Equal(tree3, []string{""}); diff != nil {
+		t.Fatalf("Found diff after remove: %s", diff)
+	}
+}
+
+func dumpPathTree(t testing.TB, root string) []string {
+	tree := make([]string, 0)
+	err := filepath.Walk(root, func(path string, _ os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		p := strings.TrimPrefix(path, root)
```

Comment:
I think this is `FileInfo.Name()` (apart from the first step, at the root... sigh...).

Reply:
Because it is a temporary name I remote the full prefix - remove just the name is problematic